src/common/serialize.rs

```diff
@@ -62,12 +53,10 @@ impl BinarySerializable for u32 {
     fn serialize(&self, writer: &mut Write) -> io::Result<usize> {
         writer.write_u32::<Endianness>(*self)
               .map(|_| 4)
-              .map_err(convert_byte_order_error)
```

Comment:
Indeed!  Always nice to have stuff do more on its own rather than having to do it yourself :+1: